drivers/adc/adc_stm32.c

```diff
@@ -832,6 +833,14 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 		__LL_ADC_CHANNEL_TO_DECIMAL_NB(LL_ADC_CHANNEL_VREFINT) == channel_id) {
 		adc_stm32_set_common_path(dev, LL_ADC_PATH_INTERNAL_VREFINT);
 	}
+#if defined(CONFIG_STM32_VBAT)
```

Comment:
Is that required ?
Obviously it will save few bytes, but can we imagine this being used w/o stm32_vbat sensor ?
Also if this to be kept, should we also use `STM32_TEMP` for the above temp related lines ?

Reply:
the LL_ADC_CHANNEL_VBAT is not available in all the stm32 mcus (unlike Temp and vRef )

Reply:
ok, then does it mean CONFIG_STM32_VBAT is not compatible with all series ?

Reply:
**stm32F1x** has no Vbat monitoring channel (but has a temperature sensor channel)